src/main/kotlin/org/jmailen/gradle/kotlinter/KotlinterPlugin.kt

```diff
@@ -20,146 +21,117 @@ import org.jmailen.gradle.kotlinter.tasks.LintTask
 class KotlinterPlugin : Plugin<Project> {
 
     val extendablePlugins = mapOf(
-            "org.jetbrains.kotlin.jvm" to KotlinJvmSourceSetResolver,
-            "kotlin-android" to AndroidSourceSetResolver)
+        "org.jetbrains.kotlin.jvm" to KotlinJvmSourceSetResolver,
+        "kotlin-android" to AndroidSourceSetResolver
+    )
 
-    override fun apply(project: Project) {
-        val kotlinterExtension = project.extensions.create("kotlinter", KotlinterExtension::class.java)
+    override fun apply(project: Project) = with(project) {
+        val kotlinterExtension = extensions.create("kotlinter", KotlinterExtension::class.java)
 
         // for known kotlin plugins, create tasks by convention.
-        val taskCreator = TaskCreator(project)
-        extendablePlugins.forEach { pluginId, sourceResolver ->
-            project.plugins.withId(pluginId) {
-
-                sourceResolver.applyToAll(project) { id, files ->
-                    taskCreator.createSourceSetTasks(id, files)
+        extendablePlugins.forEach { (pluginId, sourceResolver) ->
+            plugins.withId(pluginId) {
+
+                val lintKotlin = registerParentLintTask()
+                val formatKotlin = registerParentFormatTask()
+
+                sourceResolver.applyToAll(project) { id, resolveSources ->
+                    val ktLintParams = KtLintParams(
+                        kotlinterExtension.indentSize,
+                        kotlinterExtension.continuationIndentSize,
+                        kotlinterExtension.experimentalRules,
+                        kotlinterExtension.disabledRules,
+                        editorConfigPath()
+                    )
```

Comment:
Oh, I've started wondering if that won't be too quckly to read from extension. In case when tests don't cover such I'll make them verify that.

Reply:
Yup, that was too early. Tests didn't cover passing parameters using Kotlinter extension so I added them. 
I changed the code so the `KtLintParams` are obtained when tasks are configured.